src/obr/core/caseOrigins.py

```diff
@@ -71,17 +79,71 @@ def __init__(
 
     def init(self, path):
         if self.cache_folder and Path(self.cache_folder).exists():
-            check_output(["cp", "-r", self.cache_folder, path + "/case"])
-            return
+            # if cache folder has been cloned into before, check for new commits
+            # either way, copy from cache folder to $path/case
+            # if repo init or copying fails for some reason,
+
+            if Path(self.cache_folder + "/.git").exists():
+                repo = Repo(self.cache_folder)
+                default_remote: str = repo.git.symbolic_ref(
+                    "refs/remotes/origin/HEAD", "--short"
+                )
+                current_commit = repo.git.rev_parse("HEAD")
+                origin_branchname = default_remote.split("/")
+                # @greole: is it better to compare against self.commit or <defaultBranch>:latest ?
+                if self.commit and self.commit != current_commit:
+                    repo.git.pull(origin_branchname[0], origin_branchname[-1])
```

Comment:
I am a bit unsure about this. If `self.commit` is specified and it doesn't match the current_commit of the cache then cache repo needs to fetch and checkout the required commit. If only a branch or nothing is specified the cache needs to be on latest.

Reply:
good point!